brooklyn-server/core/src/main/java/org/apache/brooklyn/core/catalog/internal/BasicBrooklynCatalog.java

```diff
@@ -391,18 +396,61 @@ else for (String key: otherKeys) {
     }
 
     @SuppressWarnings("unchecked")
-    private void collectCatalogItems(String sourceYaml, Map<?,?> itemMetadata, List<CatalogItemDtoAbstract<?, ?>> result, Map<?,?> parentMetadata) {
+    private void collectCatalogItems(String sourceYaml, Map<String,Object> itemMetadata, List<CatalogItemDtoAbstract<?, ?>> result, Map<String,?> parentMetadata) {
 
         if (sourceYaml==null) sourceYaml = new Yaml().dump(itemMetadata);
 
-        Map<Object,Object> catalogMetadata = MutableMap.builder().putAll(parentMetadata).putAll(itemMetadata).build();
-        
+        Map<String, Object> itemEvaluatedMetadata = MutableMap.of();
+
+        itemEvaluatedMetadata.putAll(itemMetadata);
+        itemEvaluatedMetadata.remove("item");
+        itemEvaluatedMetadata.remove("items");
+
+        Object itemKey = null;
+        Object itemsKey = null;
+        if (itemMetadata.containsKey("item")) {
+            itemKey = itemMetadata.get("item");
+        }
+        if (itemMetadata.containsKey("items")) {
+            itemsKey = itemMetadata.get("items");
+        }
+
+        itemMetadata = mgmt.getConfig().getConfig(CampYamlParser.YAML_PARSER_KEY).parse(itemEvaluatedMetadata);
+
+        ExecutionContext ec = mgmt.getServerExecutionContext();
+        for (Map.Entry<String, Object> item: itemMetadata.entrySet()) {
```

Comment:
Can we use `ValueResolver.deep(true)` to get it to check if the value is a list/map, so you just have to call something like (untested code below):

    ...  = Tasks.resolveDeepValue(itemMetadata, Object.class, ec)